packages/kbn-profiling-utils/common/profiling.ts

```diff
@@ -46,15 +47,35 @@ const frameTypeDescriptions = {
   [FrameType.Perl]: 'Perl',
   [FrameType.JavaScript]: 'JavaScript',
   [FrameType.PHPJIT]: 'PHP JIT',
+  [FrameType.Error]: 'Error',
 };
 
+/**
+ * normalize the given frame type
+ * @param ft FrameType
+ * @returns FrameType
+ */
+export function normalizeFrameType(ft: FrameType): FrameType {
+  // Normalize any frame type with error bit into our uniform error variant.
+  if (((ft as number) & 0x80) != 0) {
+    return FrameType.Error;
+  }
+
+  // Guard against new / unknown frame types, rewriting them to "unsymbolized".
+  if (!(ft in frameTypeDescriptions)) {
+    return FrameType.Unsymbolized;
+  }
+
+  return ft;
+}
+
 /**
  * get frame type name
  * @param ft FrameType
  * @returns string
  */
 export function describeFrameType(ft: FrameType): string {
-  return frameTypeDescriptions[ft];
+  return frameTypeDescriptions[normalizeFrameType(ft)];
```

Comment:
Optimally we'd do this wherever the `FrameType` is first constructed / received from network, but I didn't find a good place to do that. I think strictly speaking it's a typing violation for `FrameType` to contain something that isn't a defined enum value even temporarily. 

For the moment I simply do the normalization when the value is used in a lookup table (here + the frame color lookup) because that is sufficient to prevent crashing, but if someone with more experience in the plugin code base has a better idea for where to do this, I'm all ears.